screens/UsingScreen.js

```diff
@@ -16,49 +16,60 @@ const fredVictorCoordinates = {
 // TODO: Remember to change this to 500 prior to Beta testing
 const MAXIMUM_DISTANCE = 1000000000 // meters
 
-const UsingScreen = (props) => {
 
-  const { time, userId, token } = props;
-  const { timeRemaining } = time
+class UsingScreen extends Component {
+  
+  constructor(props) {
+    super(props);
+    this.startAlarm.bind(this);
+  }
 
-  const startAlarm = () => {
-    const distance = computeDistance(fredVictorCoordinates, props.location.coords)
-    if( distance > MAXIMUM_DISTANCE ) Alert.alert("Cannot start alarm", "There are no responders within your area", [
+  startAlarm = () => {
+    const { time, userId, token } = this.props;
+    const { timeRemaining } = time
+    const distance = computeDistance(fredVictorCoordinates, this.props.location.coords)
+    if (distance > MAXIMUM_DISTANCE) Alert.alert("Cannot start alarm", "There are no responders within your area", [
       { text: 'OK' }
     ], { cancelable: false })
     else {
       Actions.alarm();
 
-      props.makeAlarmLog(userId, timeRemaining, token);
+      this.props.makeAlarmLog(userId, timeRemaining, token);
     }
   }
 
-  return (
-    <Container>
-    <Header leftButton="menu" onLeftButtonPress={() => Actions.drawerOpen()}>Using Mode</Header>
+  componentWillMount(){
+    this.props.getNumberOfAvailableResponders(this.props.userId, this.props.token);
+  }
```

Comment:
Isn't it recommended not to use componentWillMount()?
https://reactjs.org/docs/react-component.html#unsafe_componentwillmount

Reply:
Right! I forgot it's deprecated. Replaced with componentDidMount()